wallet/src/transaction/builder.rs

```diff
@@ -39,7 +40,7 @@ impl TransactionBuilder {
         G: InputGenerator,
     {
         let estimate_fee = self.settings.parameters.fees.calculate(
-            None,
+            Payload::to_certificate_slice(self.certificate.payload_data().borrow()),
             self.inputs().len() as u8,
             self.outputs().len() as u8,
         );
```

Comment:
There is a function `estimate_fee(&self) -> Value` that has not need updated (and you could reuse it here (cleaning up the code).